invenio_rdm_records/ext.py

```diff
@@ -170,6 +182,13 @@ def init_services(self, app):
             review_service=ReviewService(service_configs.record),
         )
 
+        self.records_media_files_service = RDMRecordService(
```

Comment:
this one is tricky
we set two services for record, in order to simplify components of files (if I wouldn't register a new service, I would have to modify all the file components to be aware and distinguish between `record.files` and `record.media_files`)
by having a dedicated service we can use `record.files` transparently - the service knows which files field we are referring to

Reply:
additional explanation:
this is needed since all the file services and components are tightly connected with `record.files` field.
check:
https://github.com/inveniosoftware/invenio-records-resources/blob/master/invenio_records_resources/services/files/transfer.py#L108
https://github.com/inveniosoftware/invenio-records-resources/blob/master/invenio_records_resources/services/files/service.py

thanks to this dedicated service we can reuse the files service without massive code duplication.
ping @lnielsen @slint to confirm if that was the initial design aiming for (as described in https://codimd.web.cern.ch/zEYh4vA_QyiUZqk8j-KWhw#Media-files)